### ISSUE TITLE
关于知识追踪模型最后可视化部分

### ISSUE THREAD

**Issue by WinterGan** [pykt-team/pykt-toolkit#249]
你好，我是一个知识追踪方向的新手。我想请教一下，在你们的项目中，知识追踪模型结果的热力图可视化（例如，学生知识状态随时间的变化）和题目嵌入的可视化（展示题目聚类）是如何实现的？希望能分享一下代码实现，非常感谢！

**Comment by lycyhrc** [comment]
**关于第一个问题**
>DKT模型输出是一个 [BS, seq, kc] 的3D张量（批次大小×题目序列×知识点），每个值表示学生在做完某题后对各知识点的掌握概率。

可视化方法：
- 单个学生：取 knowledge_states[student_id, :, :] 绘制热力图
- 横轴=知识点，纵轴=题目序列，颜色深浅=掌握概率
- 多个学生：循环处理每个学生，生成多个热力图(也就是论文所示)
- *如果要长度保持一致使用 knowledge_states[student_id, :N, :]* 取前N道题目

主要思路总结是，从模型输出中提取隐向量，按学生维度切片提取数据，使用seaborn.heatmap()或matplotlib.imshow()可视化

**第二个问题**
可以使用从模型embedding层提取题目嵌入向量：`model.interaction_emb.weight.detach().cpu().numpy()`，然后，使用PCA和TNSE降维，如果模型训练得好，相似的题目在原始高维嵌入空间中就会比较接近聚在一起。

**Comment by WinterGan** [comment]
> **关于第一个问题**
> 
> > DKT模型输出是一个 [BS, seq, kc] 的3D张量（批次大小×题目序列×知识点），每个值表示学生在做完某题后对各知识点的掌握概率。
> 
> 可视化方法：
> 
> * 单个学生：取 knowledge_states[student_id, :, :] 绘制热力图
> * 横轴=知识点，纵轴=题目序列，颜色深浅=掌握概率
> * 多个学生：循环处理每个学生，生成多个热力图(也就是论文所示)
> * _如果要长度保持一致使用 knowledge_states[student_id, :N, :]_ 取前N道题目
> 
> 主要思路总结是，从模型输出中提取隐向量，按学生维度切片提取数据，使用seaborn.heatmap()或matplotlib.imshow()可视化
> 
> **第二个问题** 可以使用从模型embedding层提取题目嵌入向量：`model.interaction_emb.weight.detach().cpu().numpy()`，然后，使用PCA和TNSE降维，如果模型训练得好，相似的题目在原始高维嵌入空间中就会比较接近聚在一起。

不好意思，我在尝试实现这2个功能时遇到了困难，我的代码能力有限，还没能成功。请问您是否方便在 pykt 库的 DKT 模型中，提供一个实现这两个功能的代码示例？万分感谢！
